src/query-rule-context/query-rule-context.ts

```diff
@@ -0,0 +1,34 @@
+import { Component, Input, Inject, forwardRef } from '@angular/core';
+
+import { connectQueryRules } from 'instantsearch.js/es/connectors';
+import { BaseWidget } from '../base-widget';
+import { NgAisInstantSearch } from '../instantsearch/instantsearch';
+
+type FacetValue = string | number;
+
+@Component({
+  selector: 'ais-query-rule-context',
+  template: '',
+})
+export class NgAisQueryRuleContext extends BaseWidget {
+  @Input() public trackedFilters?: {
+    [facetName: string]: (facetValues: FacetValue[]) => FacetValue[];
```

Comment:
Did we consider using the [type exported from instantsearch](https://github.com/algolia/instantsearch.js/blob/develop/src/connectors/query-rules/connectQueryRules.ts#L21)?

Reply:
I'm not sure the type can be imported if it doesn't have "typings" set, but I'll try it out

Reply:
```suggestion
    [facetName: string]: (facetValues?: FacetValue[]) => FacetValue[];
```